shared/desktop/yarn-helper/action-creator-creator.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import prettier from 'prettier'
+import path from 'path'
+import json5 from 'json5'
+import fs from 'fs'
+
+type Payload = Object
+type ErrorPayload = {
+  canError: string,
+  ...Payload,
+}
+
+type ActionNS = string
+type ActionName = string
+type ActionDesc = Payload | ErrorPayload
+
+type Actions = {[key: ActionName]: ActionDesc}
+
+type FileDesc = {
+  prelude: string, // anything to prepend to our generated file
+  actions: Actions,
+}
+
+type CompileActionFn = (ns: ActionNS, actionName: ActionName, desc: ActionDesc) => string
+
+function compile(ns: ActionNS, {prelude, actions}: FileDesc): string {
+  return `// @flow
+/* eslint-disable */
+${prelude}
+
+type _ExtractReturn<B, F: (...args: any[]) => B> = B
```

Comment:
Is that syntax correct? wouldn't it be Array<any> ?

Reply:
seems fine. based on https://hackernoon.com/redux-flow-type-getting-the-maximum-benefit-from-the-fewest-key-strokes-5c006c54ec87